`JibImageBuildServicePushTest` failing intermittently on Eclipse Jenkins

### Component

JKube Kit

### Task description

### Description
I'm observing JibImageBuildServicePushTest failing intermittently on CI

```
[ERROR] Tests run: 6, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 1.851 s <<< FAILURE! -- in org.eclipse.jkube.kit.config.service.kubernetes.JibImageBuildServicePushTest$ValidMultiplatformConfiguration
[ERROR] org.eclipse.jkube.kit.config.service.kubernetes.JibImageBuildServicePushTest$ValidMultiplatformConfiguration.pushesImage -- Time elapsed: 0.091 s <<< ERROR!
org.eclipse.jkube.kit.config.service.JKubeServiceException: Error when pushing JIB image
	at org.eclipse.jkube.kit.config.service.kubernetes.JibImageBuildService.pushSingleImage(JibImageBuildService.java:86)
	at org.eclipse.jkube.kit.config.service.AbstractImageBuildService.lambda$push$0(AbstractImageBuildService.java:41)
	at org.eclipse.jkube.kit.config.service.AbstractImageBuildService.processImage(AbstractImageBuildService.java:57)
	at org.eclipse.jkube.kit.config.service.AbstractImageBuildService.push(AbstractImageBuildService.java:41)
	at org.eclipse.jkube.kit.config.service.kubernetes.JibImageBuildServicePushTest$ValidMultiplatformConfiguration.pushValidImage(JibImageBuildServicePushTest.java:226)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at java.util.ArrayList.forEach(ArrayList.java:1259)
	at java.util.ArrayList.forEach(ArrayList.java:1259)
	at java.util.ArrayList.forEach(ArrayList.java:1259)
Caused by: org.eclipse.jkube.kit.common.JKubeException: Unable to containerize image using Jib: org.apache.http.conn.HttpHostConnectException: Connect to localhost:43691 [localhost/127.0.0.1, localhost/0:0:0:0:0:0:0:1] failed: Connection refused (Connection refused)
	at org.eclipse.jkube.kit.service.jib.JibService.containerize(JibService.java:169)
	at org.eclipse.jkube.kit.service.jib.JibService.push(JibService.java:126)
	at org.eclipse.jkube.kit.config.service.kubernetes.JibImageBuildService.pushSingleImage(JibImageBuildService.java:84)
	... 8 more
Caused by: java.util.concurrent.ExecutionException: org.apache.http.conn.HttpHostConnectException: Connect to localhost:43691 [localhost/127.0.0.1, localhost/0:0:0:0:0:0:0:1] failed: Connection refused (Connection refused)
	at com.google.common.util.concurrent.AbstractFuture.getDoneValue(AbstractFuture.java:594)
	at com.google.common.util.concurrent.AbstractFuture.get(AbstractFuture.java:573)
	at com.google.common.util.concurrent.FluentFuture$TrustedFuture.get(FluentFuture.java:91)
	at com.google.cloud.tools.jib.builder.steps.StepsRunner.lambda$pushApplicationLayers$13(StepsRunner.java:511)
	at com.google.common.util.concurrent.TrustedListenableFutureTask$TrustedFutureInterruptibleTask.runInterruptibly(TrustedListenableFutureTask.java:131)
	at com.google.common.util.concurrent.InterruptibleTask.run(InterruptibleTask.java:76)
	at com.google.common.util.concurrent.TrustedListenableFutureTask.run(TrustedListenableFutureTask.java:82)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:750)
Caused by: org.apache.http.conn.HttpHostConnectException: Connect to localhost:43691 [localhost/127.0.0.1, localhost/0:0:0:0:0:0:0:1] failed: Connection refused (Connection refused)
	at org.apache.http.impl.conn.DefaultHttpClientConnectionOperator.connect(DefaultHttpClientConnectionOperator.java:156)
	at org.apache.http.impl.conn.PoolingHttpClientConnectionManager.connect(PoolingHttpClientConnectionManager.java:376)
	at org.apache.http.impl.execchain.MainClientExec.establishRoute(MainClientExec.java:393)
	at org.apache.http.impl.execchain.MainClientExec.execute(MainClientExec.java:236)
	at org.apache.http.impl.execchain.ProtocolExec.execute(ProtocolExec.java:186)
	at org.apache.http.impl.client.InternalHttpClient.doExecute(InternalHttpClient.java:185)
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:83)
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:108)
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:56)
	at com.google.api.client.http.apache.v2.ApacheHttpRequest.execute(ApacheHttpRequest.java:73)
	at com.google.api.client.http.HttpRequest.execute(HttpRequest.java:1012)
	at com.google.cloud.tools.jib.http.FailoverHttpClient.call(FailoverHttpClient.java:349)
	at com.google.cloud.tools.jib.http.FailoverHttpClient.call(FailoverHttpClient.java:266)
	at com.google.cloud.tools.jib.registry.RegistryEndpointCaller.call(RegistryEndpointCaller.java:138)
	at com.google.cloud.tools.jib.registry.RegistryEndpointCaller.call(RegistryEndpointCaller.java:114)
	at com.google.cloud.tools.jib.registry.RegistryClient.callRegistryEndpoint(RegistryClient.java:623)
	at com.google.cloud.tools.jib.registry.RegistryClient.doBearerAuth(RegistryClient.java:318)
	at com.google.cloud.tools.jib.registry.RegistryClient.doPushBearerAuth(RegistryClient.java:308)
	at com.google.cloud.tools.jib.builder.steps.AuthenticatePushStep.call(AuthenticatePushStep.java:66)
	at com.google.cloud.tools.jib.builder.steps.AuthenticatePushStep.call(AuthenticatePushStep.java:35)
	... 6 more
Caused by: java.net.ConnectException: Connection refused (Connection refused)
	at java.net.PlainSocketImpl.socketConnect(Native Method)
	at java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:350)
	at java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:206)
	at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:188)
	at java.net.SocksSocketImpl.connect(SocksSocketImpl.java:392)
	at java.net.Socket.connect(Socket.java:607)
	at org.apache.http.conn.ssl.SSLConnectionSocketFactory.connectSocket(SSLConnectionSocketFactory.java:368)
	at org.apache.http.impl.conn.DefaultHttpClientConnectionOperator.connect(DefaultHttpClientConnectionOperator.java:142)
	... 25 more
```

I'm not able to reproduce this failure locally. We need to check why is this happening on Jenkins.

### Expected Behavior
JibImageBuildServicePushTest should pass on CI

### Acceptance Criteria
- [ ] JibImageBuildServicePushTest should pass on CI

--- a/jkube-kit/build/service/jib/src/test/java/org/eclipse/jkube/kit/service/jib/JibServiceTest.java
+++ b/jkube-kit/build/service/jib/src/test/java/org/eclipse/jkube/kit/service/jib/JibServiceTest.java
@@ -19,9 +19,6 @@ import com.google.cloud.tools.jib.api.RegistryUnauthorizedException;
 import com.google.cloud.tools.jib.api.TarImage;
 import com.google.cloud.tools.jib.api.buildplan.ImageFormat;
 import com.google.cloud.tools.jib.global.JibSystemProperties;
-import com.marcnuri.helm.jni.HelmLib;
-import com.marcnuri.helm.jni.NativeLibrary;
-import com.marcnuri.helm.jni.RepoServerOptions;
 import org.apache.commons.codec.binary.Base64;
 import org.apache.commons.io.IOUtils;
 import org.eclipse.jkube.kit.build.api.assembly.BuildDirs;
@@ -31,11 +28,11 @@ import org.eclipse.jkube.kit.common.JavaProject;
 import org.eclipse.jkube.kit.common.KitLogger;
 import org.eclipse.jkube.kit.common.RegistryConfig;
 import org.eclipse.jkube.kit.common.RegistryServerConfiguration;
+import org.eclipse.jkube.kit.common.TestOciServer;
 import org.eclipse.jkube.kit.common.assertj.ArchiveAssertions;
 import org.eclipse.jkube.kit.config.image.ImageConfiguration;
 import org.eclipse.jkube.kit.config.image.build.BuildConfiguration;
 import org.junit.jupiter.api.AfterEach;
-import org.junit.jupiter.api.BeforeAll;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.DisplayName;
 import org.junit.jupiter.api.Nested;
@@ -55,35 +52,27 @@ import static org.assertj.core.api.Assertions.assertThatThrownBy;
 
 class JibServiceTest {
 
-  private static HelmLib helmLib;
   @TempDir
   private Path tempDir;
-  private String remoteOciServer;
+  private TestOciServer remoteOciServer;
   private JibLogger jibLogger;
   private TestAuthConfigFactory testAuthConfigFactory;
   private JKubeConfiguration configuration;
   private ImageConfiguration imageConfiguration;
 
-  @BeforeAll
-  static void setUpAll() {
-    helmLib = NativeLibrary.getInstance().load();
-  }
-
   @BeforeEach
   void setUp() {
-    remoteOciServer = helmLib.RepoOciServerStart(
-      new RepoServerOptions(null, "oci-user", "oci-password")
-    ).out;
+    remoteOciServer = new TestOciServer();
     jibLogger = new JibLogger(new KitLogger.SilentLogger());
     testAuthConfigFactory = new TestAuthConfigFactory();
     configuration = JKubeConfiguration.builder()
       .pullRegistryConfig(RegistryConfig.builder().build())
       .pushRegistryConfig(RegistryConfig.builder()
-        .registry(remoteOciServer)
+        .registry(remoteOciServer.getUrl())
         .settings(Collections.singletonList(RegistryServerConfiguration.builder()
-          .id(remoteOciServer)
-          .username("oci-user")
-          .password("oci-password")
+          .id(remoteOciServer.getUrl())
+          .username(remoteOciServer.getUser())
+          .password(remoteOciServer.getPassword())
           .build()))
         .build())
       .project(JavaProject.builder()
@@ -91,11 +80,16 @@ class JibServiceTest {
         .build())
       .build();
     imageConfiguration = ImageConfiguration.builder()
-      .name(remoteOciServer + "/" + "the-image-name")
+      .name(remoteOciServer.getUrl() + "/" + "the-image-name")
       .build(BuildConfiguration.builder()
         .from("scratch")
         .build())
       .build();
+  }
+
+  @AfterEach
+  void tearDown() throws Exception {
+    remoteOciServer.close();
   }
 
   @Test
@@ -207,7 +201,7 @@ class JibServiceTest {
       try (JibService jibService = new JibService(jibLogger, testAuthConfigFactory, configuration, imageConfiguration)) {
         jibService.push();
       }
-      final HttpURLConnection connection = (HttpURLConnection) new URL("http://" + remoteOciServer + "/v2/the-image-name/tags/list")
+      final HttpURLConnection connection = (HttpURLConnection) new URL("http://" + remoteOciServer.getUrl() + "/v2/the-image-name/tags/list")
         .openConnection();
       connection.setRequestProperty("Authorization", "Basic " + Base64.encodeBase64String("oci-user:oci-password".getBytes()));
       connection.connect();
@@ -227,7 +221,7 @@ class JibServiceTest {
       try (JibService jibService = new JibService(jibLogger, testAuthConfigFactory, configuration, imageConfiguration)) {
         jibService.push();
       }
-      final HttpURLConnection connection = (HttpURLConnection) new URL("http://" + remoteOciServer + "/v2/the-image-name/tags/list")
+      final HttpURLConnection connection = (HttpURLConnection) new URL("http://" + remoteOciServer.getUrl() + "/v2/the-image-name/tags/list")
         .openConnection();
       connection.setRequestProperty("Authorization", "Basic " + Base64.encodeBase64String("oci-user:oci-password".getBytes()));
       connection.connect();
@@ -248,7 +242,7 @@ class JibServiceTest {
       try (JibService jibService = new JibService(jibLogger, testAuthConfigFactory, configuration, imageConfiguration)) {
         jibService.push();
       }
-      final HttpURLConnection connection = (HttpURLConnection) new URL("http://" + remoteOciServer + "/v2/the-image-name/manifests/latest")
+      final HttpURLConnection connection = (HttpURLConnection) new URL("http://" + remoteOciServer.getUrl() + "/v2/the-image-name/manifests/latest")
         .openConnection();
       connection.setRequestProperty("Authorization", "Basic " + Base64.encodeBase64String("oci-user:oci-password".getBytes()));
       connection.setRequestProperty("Accept", "application/vnd.docker.distribution.manifest.list.v2+json");

--- a/jkube-kit/common/pom.xml
+++ b/jkube-kit/common/pom.xml
@@ -147,6 +147,11 @@
       <groupId>com.networknt</groupId>
       <artifactId>json-schema-validator</artifactId>
     </dependency>
+    <dependency>
+      <groupId>com.marcnuri.helm-java</groupId>
+      <artifactId>helm-java</artifactId>
+      <scope>test</scope>
+    </dependency>
 
   </dependencies>
 

--- a/jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/TestOciServer.java
+++ b/jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/TestOciServer.java
@@ -1,0 +1,84 @@
+/*
+ * Copyright (c) 2019 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at:
+ *
+ *     https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.jkube.kit.common;
+
+import com.marcnuri.helm.jni.HelmLib;
+import com.marcnuri.helm.jni.NativeLibrary;
+import com.marcnuri.helm.jni.RepoServerOptions;
+import lombok.Getter;
+import org.eclipse.jkube.kit.common.util.Base64Util;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.time.Duration;
+
+import static org.eclipse.jkube.kit.common.util.AsyncUtil.await;
+import static org.eclipse.jkube.kit.common.util.AsyncUtil.get;
+
+@Getter
+public class TestOciServer implements Closeable {
+
+  private static final Duration TIMEOUT = Duration.ofSeconds(5);
+  private static final String DEFAULT_USER = "oci-user";
+  private static final String DEFAULT_PASSWORD = "oci-password";
+
+  // initialization-on-demand holder
+  private static final class HelmLibHolder {
+    private static final HelmLib INSTANCE = NativeLibrary.getInstance().load();
+  }
+
+  private static HelmLib getHelmLib() {
+    return HelmLibHolder.INSTANCE;
+  }
+
+  private final String user;
+  private final String password;
+  private final String url;
+
+  public TestOciServer() {
+    this(DEFAULT_USER, DEFAULT_PASSWORD);
+  }
+
+  public TestOciServer(String user, String password) {
+    this.user = user;
+    this.password = password;
+    url = get(await(this::startServer).apply(this::waitForServer), TIMEOUT);
+  }
+
+  @Override
+  public void close() throws IOException {
+    // No effect yet https://github.com/manusa/helm-java/blob/f44a88ed1ad351b2b5a00b5e735deb5cb35b32f7/native/internal/helm/repotest.go#L138
+    getHelmLib().RepoServerStop(url);
+  }
+
+  private String startServer() {
+    return getHelmLib().RepoOciServerStart(
+      new RepoServerOptions(null, user, password)
+    ).out;
+  }
+
+  private boolean waitForServer(String serverUrl) {
+    try {
+      final HttpURLConnection connection = (HttpURLConnection) new URL("http://" + serverUrl + "/v2/")
+        .openConnection();
+      connection.setRequestProperty("Authorization", "Basic " + Base64Util.encodeToString(String.join(":", user, password)));
+      connection.connect();
+      return connection.getResponseCode() == HttpURLConnection.HTTP_OK;
+    } catch (IOException e) {
+      return false;
+    }
+  }
+}

--- a/jkube-kit/config/service/src/test/java/org/eclipse/jkube/kit/config/service/kubernetes/JibImageBuildServicePushTest.java
+++ b/jkube-kit/config/service/src/test/java/org/eclipse/jkube/kit/config/service/kubernetes/JibImageBuildServicePushTest.java
@@ -18,9 +18,6 @@ import com.google.cloud.tools.jib.api.Jib;
 import com.google.cloud.tools.jib.api.TarImage;
 import com.google.cloud.tools.jib.api.buildplan.ImageFormat;
 import com.google.cloud.tools.jib.global.JibSystemProperties;
-import com.marcnuri.helm.jni.HelmLib;
-import com.marcnuri.helm.jni.NativeLibrary;
-import com.marcnuri.helm.jni.RepoServerOptions;
 import org.apache.commons.codec.binary.Base64;
 import org.apache.commons.io.IOUtils;
 import org.eclipse.jkube.kit.common.JKubeConfiguration;
@@ -29,6 +26,7 @@ import org.eclipse.jkube.kit.common.JavaProject;
 import org.eclipse.jkube.kit.common.KitLogger;
 import org.eclipse.jkube.kit.common.RegistryConfig;
 import org.eclipse.jkube.kit.common.RegistryServerConfiguration;
+import org.eclipse.jkube.kit.common.TestOciServer;
 import org.eclipse.jkube.kit.config.image.ImageConfiguration;
 import org.eclipse.jkube.kit.config.image.build.BuildConfiguration;
 import org.eclipse.jkube.kit.config.resource.RuntimeMode;
@@ -36,26 +34,20 @@ import org.eclipse.jkube.kit.config.service.BuildServiceConfig;
 import org.eclipse.jkube.kit.config.service.JKubeServiceException;
 import org.eclipse.jkube.kit.config.service.JKubeServiceHub;
 import org.junit.jupiter.api.AfterEach;
-import org.junit.jupiter.api.BeforeAll;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.DisplayName;
 import org.junit.jupiter.api.Nested;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.io.TempDir;
 
-import java.io.IOException;
 import java.net.HttpURLConnection;
 import java.net.URL;
 import java.nio.charset.StandardCharsets;
 import java.nio.file.Path;
 import java.util.Collections;
-import java.util.concurrent.ExecutionException;
-import java.util.concurrent.TimeUnit;
-import java.util.concurrent.TimeoutException;
 
 import static org.assertj.core.api.Assertions.assertThat;
 import static org.assertj.core.api.Assertions.assertThatThrownBy;
-import static org.eclipse.jkube.kit.common.util.AsyncUtil.await;
 import static org.mockito.ArgumentMatchers.anyString;
 import static org.mockito.Mockito.never;
 import static org.mockito.Mockito.spy;
@@ -64,22 +56,15 @@ import static org.mockito.Mockito.verify;
 
 class JibImageBuildServicePushTest {
 
-  private static HelmLib helmLib;
-
   @TempDir
   private Path temporaryFolder;
 
   private boolean sendCredentialsOverHttp;
-  private String remoteOciServer;
+  private TestOciServer remoteOciServer;
 
   private KitLogger logger;
   private JKubeServiceHub jKubeServiceHub;
   private ImageConfiguration imageConfiguration;
-
-  @BeforeAll
-  static void setUpAll() {
-    helmLib = NativeLibrary.getInstance().load();
-  }
 
   @BeforeEach
   void setUp() throws Exception {
@@ -89,10 +74,7 @@ class JibImageBuildServicePushTest {
     logger = spy(new KitLogger.SilentLogger());
 
     // Setup OCI server
-    remoteOciServer = helmLib.RepoOciServerStart(
-      new RepoServerOptions(null, "oci-user", "oci-password")
-    ).out;
-    waitForOciServerToAcceptConnections();
+    remoteOciServer = new TestOciServer();
 
     // Configure OCI server in image and JKube
     imageConfiguration = ImageConfiguration.builder()
@@ -100,7 +82,7 @@ class JibImageBuildServicePushTest {
       .build(BuildConfiguration.builder()
         .from("scratch")
         .build())
-      .registry(remoteOciServer)
+      .registry(remoteOciServer.getUrl())
       .build();
     jKubeServiceHub = JKubeServiceHub.builder()
       .log(logger)
@@ -110,12 +92,12 @@ class JibImageBuildServicePushTest {
         .project(JavaProject.builder().baseDirectory(temporaryFolder.toFile()).build())
         .pullRegistryConfig(RegistryConfig.builder().settings(Collections.emptyList()).build())
         .pushRegistryConfig(RegistryConfig.builder()
-          .registry(remoteOciServer)
+          .registry(remoteOciServer.getUrl())
           .settings(Collections.singletonList(
             RegistryServerConfiguration.builder()
-              .id(remoteOciServer)
-              .username("oci-user")
-              .password("oci-password")
+              .id(remoteOciServer.getUrl())
+              .username(remoteOciServer.getUser())
+              .password(remoteOciServer.getPassword())
               .build()))
           .passwordDecryptionMethod(s -> s)
           .build())
@@ -125,7 +107,7 @@ class JibImageBuildServicePushTest {
     // Build fake container image
     final Path tarPath = temporaryFolder
       .resolve("localhost")
-      .resolve(remoteOciServer.split(":")[1])
+      .resolve(remoteOciServer.getUrl().split(":")[1])
       .resolve("test")
       .resolve("test-image")
       .resolve("0.0.1")
@@ -139,24 +121,9 @@ class JibImageBuildServicePushTest {
       );
   }
 
-  private void waitForOciServerToAcceptConnections() throws ExecutionException, InterruptedException, TimeoutException {
-    await(() -> {
-      try {
-        final HttpURLConnection connection = (HttpURLConnection) new URL("http://" + remoteOciServer + "/v2/")
-          .openConnection();
-        connection.setRequestProperty("Authorization", "Basic " + Base64.encodeBase64String("oci-user:oci-password".getBytes()));
-        connection.connect();
-        return connection.getResponseCode() == HttpURLConnection.HTTP_OK;
-      } catch (IOException e) {
-        return false;
-      }
-    })
-      .apply(t -> t)
-      .get(5, TimeUnit.SECONDS);
-  }
-
   @AfterEach
-  void tearDown() {
+  void tearDown() throws Exception {
+    remoteOciServer.close();
     if (!sendCredentialsOverHttp) {
       System.clearProperty(JibSystemProperties.SEND_CREDENTIALS_OVER_HTTP);
     }
@@ -214,12 +181,12 @@ class JibImageBuildServicePushTest {
     @Test
     void logsImagePush() {
       verify(logger, times(1))
-        .info("Pushing image: %s", remoteOciServer + "/test/test-image:0.0.1");
+        .info("Pushing image: %s", remoteOciServer.getUrl() + "/test/test-image:0.0.1");
     }
 
     @Test
     void pushesImage() throws Exception {
-      final HttpURLConnection connection = (HttpURLConnection) new URL("http://" + remoteOciServer + "/v2/test/test-image/tags/list")
+      final HttpURLConnection connection = (HttpURLConnection) new URL("http://" + remoteOciServer.getUrl() + "/v2/test/test-image/tags/list")
         .openConnection();
       connection.setRequestProperty("Authorization", "Basic " + Base64.encodeBase64String("oci-user:oci-password".getBytes()));
       connection.connect();
@@ -250,12 +217,12 @@ class JibImageBuildServicePushTest {
     @Test
     void logsImagePush() {
       verify(logger, times(1))
-        .info("Pushing image: %s", remoteOciServer + "/test/test-image:multiplatform");
+        .info("Pushing image: %s", remoteOciServer.getUrl() + "/test/test-image:multiplatform");
     }
 
     @Test
     void pushesImage() throws Exception {
-      final HttpURLConnection connection = (HttpURLConnection) new URL("http://" + remoteOciServer + "/v2/test/test-image/tags/list")
+      final HttpURLConnection connection = (HttpURLConnection) new URL("http://" + remoteOciServer.getUrl() + "/v2/test/test-image/tags/list")
         .openConnection();
       connection.setRequestProperty("Authorization", "Basic " + Base64.encodeBase64String("oci-user:oci-password".getBytes()));
       connection.connect();
@@ -266,7 +233,7 @@ class JibImageBuildServicePushTest {
 
     @Test
     void pushedImageManifestIsMultiplatform() throws Exception {
-      final HttpURLConnection connection = (HttpURLConnection) new URL("http://" + remoteOciServer + "/v2/test/test-image/manifests/multiplatform")
+      final HttpURLConnection connection = (HttpURLConnection) new URL("http://" + remoteOciServer.getUrl() + "/v2/test/test-image/manifests/multiplatform")
         .openConnection();
       connection.setRequestProperty("Authorization", "Basic " + Base64.encodeBase64String("oci-user:oci-password".getBytes()));
       connection.setRequestProperty("Accept", "application/vnd.docker.distribution.manifest.list.v2+json");

--- a/jkube-kit/config/service/src/test/java/org/eclipse/jkube/kit/config/service/kubernetes/JibImageBuildServicePushTest.java
+++ b/jkube-kit/config/service/src/test/java/org/eclipse/jkube/kit/config/service/kubernetes/JibImageBuildServicePushTest.java
@@ -43,14 +43,19 @@ import org.junit.jupiter.api.Nested;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.io.TempDir;
 
+import java.io.IOException;
 import java.net.HttpURLConnection;
 import java.net.URL;
 import java.nio.charset.StandardCharsets;
 import java.nio.file.Path;
 import java.util.Collections;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
 
 import static org.assertj.core.api.Assertions.assertThat;
 import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.eclipse.jkube.kit.common.util.AsyncUtil.await;
 import static org.mockito.ArgumentMatchers.anyString;
 import static org.mockito.Mockito.never;
 import static org.mockito.Mockito.spy;
@@ -66,7 +71,6 @@ class JibImageBuildServicePushTest {
 
   private boolean sendCredentialsOverHttp;
   private String remoteOciServer;
-
 
   private KitLogger logger;
   private JKubeServiceHub jKubeServiceHub;
@@ -88,6 +92,7 @@ class JibImageBuildServicePushTest {
     remoteOciServer = helmLib.RepoOciServerStart(
       new RepoServerOptions(null, "oci-user", "oci-password")
     ).out;
+    waitForOciServerToAcceptConnections();
 
     // Configure OCI server in image and JKube
     imageConfiguration = ImageConfiguration.builder()
@@ -132,6 +137,22 @@ class JibImageBuildServicePushTest {
         .at(tarPath)
         .named(imageConfiguration.getName()))
       );
+  }
+
+  private void waitForOciServerToAcceptConnections() throws ExecutionException, InterruptedException, TimeoutException {
+    await(() -> {
+      try {
+        final HttpURLConnection connection = (HttpURLConnection) new URL("http://" + remoteOciServer + "/v2/")
+          .openConnection();
+        connection.setRequestProperty("Authorization", "Basic " + Base64.encodeBase64String("oci-user:oci-password".getBytes()));
+        connection.connect();
+        return connection.getResponseCode() == HttpURLConnection.HTTP_OK;
+      } catch (IOException e) {
+        return false;
+      }
+    })
+      .apply(t -> t)
+      .get(5, TimeUnit.SECONDS);
   }
 
   @AfterEach

